build.ps1

```diff
@@ -59,4 +59,6 @@ bazel build `-`-experimental_execution_log_file ${ARTIFACT_DIRS}/logs/build_exec
 
 bazel shutdown
 
-bazel test `-`-experimental_execution_log_file ${ARTIFACT_DIRS}/logs/test_execution_windows.log //...
+if ($env:SKIP_TESTS = "False") {
```

Comment:
Maybe worth adding a comment here or in the yaml file documenting that
1. We only do this for release builds where the PR (should) run exactly the same tests as the build on main.
2. The reason why release builds don’t benefit from caching on Windows is because we have to cache per working directory and different pipelines run in different working directories.

Reply:
Added comments on the `ci/build-windows.yml` side; I think it makes more sense to have the comments near the decision logic.